src/buf/fixed/registry.rs

```diff
@@ -0,0 +1,114 @@
+    /// buffers from the input. Any items in excess of that amount are silently
+    /// dropped, unless the input iterator produces the vectors lazily.
+    ///
+    /// [`UIO_MAXIOV`]: libc::UIO_MAXIOV
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use tokio_uring::buf::fixed::FixedBufRegistry;
+    /// use std::iter;
+    ///
+    /// let registry = FixedBufRegistry::new(iter::repeat(vec![0; 4096]).take(10));
+    /// ```
+    pub fn new(bufs: impl IntoIterator<Item = Vec<u8>>) -> Self {
+        FixedBufRegistry {
+            inner: Rc::new(RefCell::new(FixedBuffers::new(bufs.into_iter()))),
+        }
+    }
+
+    /// Registers the buffers with the kernel.
+    ///
+    /// This method must be called in the context of a `tokio-uring` runtime.
+    /// The registration persists for the lifetime of the runtime, unless
+    /// revoked by the [`unregister`] method. Dropping the
+    /// `FixedBufRegistry` instance this method has been called on does not revoke
+    /// the registration or deallocate the buffers.
+    ///
+    /// [`unregister`]: Self::unregister
+    ///
+    /// This call can be blocked in the kernel to complete any operations
+    /// in-flight on the same `io-uring` instance. The application is
+    /// recommended to register buffers before starting any I/O operations.
+    ///
+    /// # Errors
+    ///
+    /// If a collection of buffers is currently registered in the context
+    /// of the `tokio-uring` runtime this call is made in, the function returns
+    /// an error.
+    pub fn register(&self) -> io::Result<()> {
+        driver::register_buffers(&self.inner)
+    }
+
+    /// Unregisters this collection of buffers.
+    ///
+    /// This method must be called in the context of a `tokio-uring` runtime,
+    /// where the buffers should have been previously registered.
+    ///
+    /// This operation invalidates any `FixedBuf` handles checked out from
+    /// this registry instance. Continued use of such handles in I/O
+    /// operations may result in an error.
+    ///
+    /// # Errors
+    ///
+    /// If another collection of buffers is currently registered in the context
+    /// of the `tokio-uring` runtime this call is made in, the function returns
+    /// an error. Calling `unregister` when no `FixedBufRegistry` is currently
+    /// registered on this runtime also returns an error.
+    pub fn unregister(&self) -> io::Result<()> {
+        driver::unregister_buffers(&self.inner)
+    }
+
+    /// Returns a buffer identified by the specified index for use by the
+    /// application, unless the buffer is already in use.
+    ///
+    /// The buffer is released to be available again once the
+    /// returned `FixedBuf` handle has been dropped. An I/O operation
+    /// using the buffer takes ownership of it and returns it once completed,
+    /// preventing shared use of the buffer while the operation is in flight.
+    pub fn check_out(&self, index: usize) -> Option<FixedBuf> {
```

Comment:
Perhaps this should be renamed to just `get`: the API user is not much interested in the internal mechanics of checking buffers in and out.

Reply:
I could see either way. I like specific names like check_out, but over the long haul, packages like tokio seem to go with short meaning names, even if those names are used in other modules for other purposes too. I'll let you and @Noah-Kennedy figure this out.

Reply:
I recall now why I decided against `get`: this name is commonly given to idempotent operations, but here you can't "get" the buffer under the same index a second time unless you drop it first.

Reply:
Okay. Good enough reason for me. Also, I have to remind myself too that none of this is cast in store. Getting it in and letting people use it is the most important thing. Everyone should know APIs can change this year and probably into next year.